### ISSUE TITLE
Issues installing with python v3.12

### ISSUE THREAD

**Issue by rjsmethurst** [danielrd6/ifscube#23]
numpy.distutils has been depreciated with Python v3.12 and the setup.py now doesn't run. Changing the first line of code in setup.py from:

`from numpy.distutils.core import Extension, setup`

to:

`from setuptools import Extension, setup`

(as recommended by [setuptools documentation](https://setuptools.pypa.io/en/latest/userguide/quickstart.html)) let's the code run, but then the Extension module runs into issue with the fortran files and you get this error:

```
building 'ifscube.elprofile' extension
error: unknown file type '.f90' (from 'ifscube/profiles.f90')
```

**Comment by danielrd6** [comment]
Indeed, the `distutils` deprecation requires a significant change in the packaging and installation procedure. I will work on this issue during the coming weeks, but until a solution is implemented we will have to keep using IFSCube in a 3.11 environment.

**Comment by danielrd6** [comment]
All Fortran code, which required distutils for compiling and linking the functions describing the spectral feature profiles, has been removed. The current version of IFSCube uses Numba for anything that was written in Fortran in previous versions. This will ensure greater compatibility in the foreseeable future, and reduce the time required for maintaining the code.
